troughs_and_waves/gaussian/tests/test_gaussian_factory.py

```diff
@@ -0,0 +1,30 @@
+#  (C) Copyright 2010-2020 Enthought, Inc., Austin, TX
+#  All rights reserved.
+
+import unittest
+
+from troughs_and_waves.gaussian.gaussian import Gaussian
+from troughs_and_waves.gaussian.gaussian_model import GaussianModel
+
+from troughs_and_waves.troughs_and_waves_plugin import TroughsAndWavesPlugin
+
+
+class DataSourceFactoryTestMixin(unittest.TestCase):
+    def setUp(self):
+        self.plugin = TroughsAndWavesPlugin()
+        self.factory = self.plugin.data_source_factories[0]
+
+    def test_initialization(self):
+        self.assertNotEqual(self.factory.id, "")
+        self.assertEqual(self.factory.plugin_id, self.plugin.id)
+
+    def test_create_model(self):
+        model = self.factory.create_model({})
+        self.assertIsInstance(model, GaussianModel)
+
+        model = self.factory.create_model()
+        self.assertIsInstance(model, GaussianModel)
+
+    def test_create_data_source(self):
+        ds = self.factory.create_data_source()
+        self.assertIsInstance(ds, Gaussian)
```

Comment:
Apologies for not spotting this previously, but generally we don't need to make dedicated `TestCase` objects for the BDSS factory subclasses, since they really just provide a load of boiler plate code. Nearly all functionalities of the abstract methods that we are checking here should already be taken care of in the core BDSS package.

In order to test that the `BaseDataSouceFactory` subclasses are working properly, we can normally just include them alongside the rest of the `BaseDataSource` and `BaseDataSourceModel` tests (see other comment). We can also do the same for the other data source classes contributed in this plugin (`CircularWave`, `GaussValley`).

_Minor note:_ all of these tests are also dependant on the order of `TroughsAndWavesPlugin.get_factory_classes`, which isn't actually important for the functionality of the plugin and if it changes we may end up with a load of broken tests. I know other BDSS plugin repositories have similar `setUp` test methods, that were (and are no longer needed) work arounds to obtain a `plugin` reference in order to instantiate factory objects. Unfortunately this makes things a bit confusing, but it is not a good practice to follow.